sources/packages/api/src/esdc-integration/esdc-file-handler.ts

```diff
@@ -0,0 +1,45 @@
+import { OfferingIntensity } from "../database/entities";
+import {
+  CreateRequestFileNameResult,
+  DATE_FORMAT,
+  NUMBER_FILLER,
+} from "./models/esdc-integration.model";
+import { StringBuilder } from "../utilities/string-builder";
+import { ConfigService } from "../services";
+import { ESDCIntegrationConfig } from "../types";
+export abstract class ESDCFileHandler {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(config: ConfigService) {
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+  /**
+   * Expected file name of the request file.
+   * @param filenameCode different files has different filename codes, to be created.
+   * @param OfferingIntensity offering intensity of the application
+   *  where file is requested.
+   * @param nextSequenceNumber
+   * @returns Full file path of the file to be saved on the SFTP.
+   */
+  async createRequestFileName(
+    filenameCode: string,
+    nextSequenceNumber: number,
+    offeringIntensity?: OfferingIntensity,
+  ): Promise<CreateRequestFileNameResult> {
+    const fileNameArray = new StringBuilder();
+    fileNameArray.append(`${this.esdcConfig.environmentCode}${filenameCode}`);
+    if (offeringIntensity && OfferingIntensity.partTime === offeringIntensity) {
```

Comment:
I believe that there is no need to check if _offeringIntensity_ is defined. It could be just as below.
```typescript
if (offeringIntensity === OfferingIntensity.partTime)
```